### ISSUE TITLE
[Bug]: swc-react picker is not working on an overlay in mobile only in Firefly

### ISSUE THREAD

**Issue by Rajdeepc** [adobe/spectrum-web-components#3834]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

Picker

### Expected behavior

[Issue in Mobile]
The picker options should pop up from the bottom of the screen in mobile and not from the bottom of the dialog/overlay element.
![pickerissue](https://github.com/adobe/spectrum-web-components/assets/8790510/bb9dd82b-a4b8-4d50-9f17-d59fc3be1568)

### Actual behavior

The picker should pop up as tray when triggered from a dialog/overlay

![pickeroriginal](https://github.com/adobe/spectrum-web-components/assets/8790510/8164e4be-20ac-4f19-8e1f-c9a8c839a202)

### How can we reproduce this issue?

1. Go to https://firefly-stage-pr.corp.adobe.com/PR-3112/ in mobile
2. Login
3. Click on user -> preferences
4. Select from a picker
5. See UI breaks

### Sample code that illustrates the problem

![Screenshot 2023-11-28 at 10 52 23 PM](https://github.com/adobe/spectrum-web-components/assets/8790510/19389603-6b84-4831-8785-af8ca1fdaea3)

[OPEN IN MOBILE DEVICE]

REPRO: https://studio.webcomponents.dev/edit/RSDikStPmUPSioVpCsYb/src/index.ts?branch=dialogandpicker%40j2dknEdsGAeqDKWaGlfs7jIh72k1&p=stories

**Comment by jianliao** [comment]
Hello @Rajdeepc,

I've attached a screen recording below for your reference. It appears that this issue only manifests during the first access; on the second attempt, the tray behaves as expected. I conducted a visual comparison of the DOM tree between the first and second accesses but didn't notice any differences. Also, this issue doesn't seem to be related to the React wrapper, as the initial DOM elements were correctly in place and upgraded without issue.

https://github.com/adobe/spectrum-web-components/assets/1207520/400cb416-bd97-40b3-b88e-95604b238610
